engine.py

```diff
@@ -198,6 +209,113 @@ def log_warning(self, msg):
         print "Shotgun Warning: %s" % msg
 
     ############################################################################
+    # panel interfaces
+    ############################################################################
+
+    def get_panel_info(self, requested_panel_id):
+        """Get info dict for the panel with the supplied id."""
+
+        for (panel_id, panel_dict) in self.panels.items():
+            if not panel_id == requested_panel_id:
+                continue
+
+            # for now, we set a state flag to tell show_panel that we really
+            # just need an instance of the widget. this is what the panel
+            # interfaces will call to display their contents.
+            self._panel_info_request = True
+            self.log_debug("Retrieving panel widget for %s" % panel_id)
+            panel_info = panel_dict['callback']()
```

Comment:
maybe a comment here explaining what happens here, how the callback works etc could be helpful.